src/battle_controllers.c

```diff
@@ -0,0 +1,1551 @@
+    gBattleBuffersTransferData[5] = size;
+    gBattleBuffersTransferData[6] = (size & 0xFF00) >> 8;
+    for (i = 0; i < size; i++)
+        gBattleBuffersTransferData[7 + i] = *(u8*)(data++);
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, size + 7);
+}
+
+void EmitPlayBGM(u8 bufferId, u16 songId, void *unusedDumbDataParameter)
+{
+    s32 i;
+
+    gBattleBuffersTransferData[0] = CONTROLLER_31;
+    gBattleBuffersTransferData[1] = songId;
+    gBattleBuffersTransferData[2] = (songId & 0xFF00) >> 8;
+    for (i = 0; i < songId; i++) // ????
+        gBattleBuffersTransferData[3 + i] = *(u8*)(unusedDumbDataParameter++);
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, songId + 3);
+}
+
+void EmitCmd32(u8 bufferId, u16 size, void *data)
+{
+    s32 i;
+
+    gBattleBuffersTransferData[0] = CONTROLLER_32;
+    gBattleBuffersTransferData[1] = size;
+    gBattleBuffersTransferData[2] = (size & 0xFF00) >> 8;
+    for (i = 0; i < size; i++)
+        gBattleBuffersTransferData[3 + i] = *(u8*)(data++);
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, size + 3);
+}
+
+void EmitCmd33(u8 bufferId, u8 arg1, u16 arg2)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_33;
+    gBattleBuffersTransferData[1] = arg1;
+    gBattleBuffersTransferData[2] = arg2;
+    gBattleBuffersTransferData[3] = (arg2 & 0xFF00) >> 8;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd34(u8 bufferId, u8 b, u8 *c)
+{
+    s32 i;
+
+    gBattleBuffersTransferData[0] = CONTROLLER_34;
+    gBattleBuffersTransferData[1] = b;
+    for (i = 0; i < 3; i++)
+        gBattleBuffersTransferData[2 + i] = c[i];
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 5);
+}
+
+void EmitCmd35(u8 bufferId, u16 b)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_35;
+    gBattleBuffersTransferData[1] = b;
+    gBattleBuffersTransferData[2] = (b & 0xFF00) >> 8;
+    gBattleBuffersTransferData[3] = 0;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd36(u8 bufferId, u16 b)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_36;
+    gBattleBuffersTransferData[1] = b;
+    gBattleBuffersTransferData[2] = (b & 0xFF00) >> 8;
+    gBattleBuffersTransferData[3] = 0;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd37(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_37;
+    gBattleBuffersTransferData[1] = 37;
+    gBattleBuffersTransferData[2] = 37;
+    gBattleBuffersTransferData[3] = 37;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd38(u8 bufferId, u8 b)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_38;
+    gBattleBuffersTransferData[1] = b;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 2);
+}
+
+void EmitCmd39(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_39;
+    gBattleBuffersTransferData[1] = 39;
```

Comment:
Shouldn't these just be use CONTROLLER_39 enum? (Same goes for all instances where the value is repeated four times, seemingly as filler)

Reply:
Yeah, it's possible they used filler to align the buffer to at least 4 bytes. I wanted to check if the seemingly filler bytes are used by any controllers first tho

Reply:
It seems obvious to me that these are just repeats of the enum value, regardless of their final usage, so it would make sense to me to make those also use the value from the enum.